README.md

```diff
@@ -1,126 +1,241 @@
+```
+Your download path may vary.
+- If you are in an Ubuntu docker container, omit `sudo` because you run as `root` by default.
 
-**A Warning On Parallel Compilation Jobs (`-j` flag)**: When building C/C++ software often the build is performed in parallel via a command such as `make -j $(nproc)` which uses the number of CPU cores as the number of compilation jobs to perform simultaneously. However, be aware that some compilation units (.cpp files) in Leap are extremely complex and will consume nearly 4GB of memory to compile. You may need to reduce the level of parallelization depending on the amount of memory on your build host. e.g. instead of `make -j $(nproc)` run `make -j2`. Failures due to memory exhaustion will typically but not always manifest as compiler crashes.
+Finally, verify Leap was installed correctly.
+```bash
+nodeos --full-version
+```
+You should see a [semantic version](https://semver.org) string print out followed by a `git` commit hash with no errors. For example:
+```
+v3.1.2-0b64f879e3ebe2e4df09d2e62f1fc164cc1125d1
+```
 
-Generally we recommend performing what we refer to as a "pinned build" which ensures the compiler and boost version remain the same between builds of different Leap versions (Leap requires these versions remain the same otherwise its state needs to be repopulated from a portable snapshot).
+## Source Installation
+You can also build and install Leap from source.
+
+### Prerequisites
+Recent Ubuntu LTS releases are the only operating systems that we support. Other Unix derivatives such as macOS are tended to on a best-effort basis and may not be full featured. If you aren't using Ubuntu, please visit [this page](./docs/00_install/01_build-from-source/00_build-unsupported-os.md) to explore your options.
+
+Notable requirements to build are:
+- C++17 compiler and standard library
+- boost 1.67+
+- CMake 3.8+
+- LLVM 7 - 11 - for Linux only
+  - newer versions do not work
+- openssl 1.1+
+- libcurl
+- curl
+- libusb
+- git
+- GMP
+- Python 3
+- zlib
+
+If you don't already have the Leap repo cloned to your computer, [open a terminal](https://itsfoss.com/open-terminal-ubuntu) and navigate to the folder you want the Leap source code to live in.
+```bash
+cd ~/Downloads
+```
+Clone Leap using HTTPS...
+```bash
+git clone --recursive https://github.com/AntelopeIO/leap.git
+```
+...or SSH.
+```bash
+git clone --recursive git@github.com:AntelopeIO/leap.git
+```
+An HTTPS and SSH clone both result in the same thing, a folder named `leap` containg our source code, it doesn't matter which one you use. Navigate into that folder.
+```bash
+cd leap
+```
+Choose which [release tag](https://github.com/AntelopeIO/leap/releases) or [branch](#branches) you would like to build and check it out. If you don't know, use the [latest release](https://github.com/AntelopeIO/leap/releases/latest). For example, if you want to build release 3.1.2, you would check it out using its tag.
+```bash
+git checkout v3.1.2
```

Comment:
Is it worth using a non-existent version as the example version? So if someone blindly copy-pastes it over it errors out.

Reply:
Yes, I like that idea.

Reply:
Done, used `v0.0.0`.